opentelemetry-api/src/opentelemetry/util/_providers.py

```diff
@@ -12,33 +12,19 @@
 # See the License for the specific language governing permissions and
```

Comment:
Didn't we decide on not having a `util` folder at all? None of these functions we want to expose to users as a "utility" so why have them under the namespace? The only function that we can argue that we might want to expose is `time_ns` but from [this pr](https://github.com/open-telemetry/opentelemetry-python/pull/1602) I don't think that's the case either.

Reply:
This PR only does not include changes from other PRs. #1602 does, I have marked #1602 as a draft while we get this one merged first.

Reply:
I'll move `types` to `opentelemetry` and remove `util` as a namespace in `opentelemetry-api` as agreed in the SIG @lzchen 
:+1:

Reply:
Ok, after trying this I don't think moving `types` outside of `utils` is a good idea. `types` is just a collection of classes that are used to provide type hints, moving them to the root of `opentelemetry-api` makes them semantic peers of actual Opentelemetry stuff like propagators. Also, there is already a `types` module in Python, when I move this outside, `pylint` goes crazy. At least when `types` is in `utils` it shows that is more of convenience code and not feature-critical.

This highlights a fundamental issue, so far we have been using `opentelemetry-api` to include the actual API stuff and any other utilitarian code that opentelemetry-related packages (like instrumentations, for example) may need because `opentelemetry-api` is the only dependency that is common to all of them. I think we want to keep `opentelemetry-api` with the minimal amount of code possible, hopefully only with stuff that is defined in the spec. On the other hand, repeated code usually ends up being placed in some `util` folder somewhere.

I say that we keep it as it is for now and we discuss this more fundamental issue later.